migrations/1686291754032-copy-legacy-tools-to-ctl.js

```diff
@@ -0,0 +1,374 @@
+	'context_external_tool0906202311484',
+	new mongoose.Schema(
+		{
+			_id: { type: Schema.Types.ObjectId, required: true },
+			schoolTool: { type: Schema.Types.ObjectId, ref: 'schoolExternalTool', required: true },
+			contextId: { type: Schema.Types.ObjectId, required: true },
+			contextType: { type: String, enum: ['course'], required: true },
+			contextToolName: { type: String },
+			parameters: [{ type: { key: { type: String }, value: { type: String } } }],
+			toolVersion: { type: Number, required: true },
+			createdAt: { type: Date, default: Date.now },
+			updatedAt: { type: Date, default: Date.now },
+		},
+		{
+			timestamps: true,
+		}
+	),
+	'context_external_tools'
+);
+
+const Course = mongoose.model(
+	'course0906202311485',
+	new mongoose.Schema(
+		{
+			_id: { type: Schema.Types.ObjectId, required: true },
+			school: { type: Schema.Types.ObjectId, required: true },
+		},
+		{
+			timestamps: true,
+		}
+	),
+	'courses'
+);
+
+function mapToExternalToolParameter(ltiToolTemplate) {
+	return ltiToolTemplate.customs.map((parameter) => {
+		return {
+			name: parameter.key,
+			displayName: parameter.key,
+			description: '',
+			default: '',
+			regex: '',
+			regexComment: '',
+			scope: 'context',
+			location: 'body',
+			type: 'string',
+			isOptional: true,
+		};
+	});
+}
+
+function mapToCustomParameterEntry(externalToolParameters, ltiToolCustomes) {
+	return externalToolParameters.map((parameter, index) => {
+		return {
+			name: parameter.name,
+			value: ltiToolCustomes[index].value,
+		};
+	});
+}
+
+function toolConfigMapper(ltiToolTemplate) {
+	let toolConfig = {
+		config_baseUrl: ltiToolTemplate.url,
+		config_type: 'basic',
+	};
+
+	if (ltiToolTemplate.oAuthClientId) {
+		toolConfig = {
+			...toolConfig,
+			config_type: 'oauth2',
+			config_clientId: ltiToolTemplate.oAuthClientId,
+			config_skipConsent: ltiToolTemplate.skipConsent,
+		};
+	} else if (ltiToolTemplate.key || ltiToolTemplate.key !== 'none') {
+		toolConfig = {
+			...toolConfig,
+			config_type: 'lti11',
+			config_key: ltiToolTemplate.key,
+			config_secret: ltiToolTemplate.secret,
+			config_ressource_link_id: ltiToolTemplate.ressource_link_id,
+			config_lti_message_type: ltiToolTemplate.lti_message_type,
+			config_privacy_permission: ltiToolTemplate.privacy_permission,
+		};
+	}
+
+	return toolConfig;
+}
+
+function mapToExternalTool(ltiToolTemplate) {
+	return {
+		name: ltiToolTemplate.name,
+		url: ltiToolTemplate.url,
+		logoUrl: ltiToolTemplate.logo_url,
+		parameters: mapToExternalToolParameter(ltiToolTemplate),
+		isHidden: ltiToolTemplate.isHidden,
+		openNewTab: ltiToolTemplate.openNewTab,
+		version: 1,
+		...toolConfigMapper(ltiToolTemplate),
+	};
+}
+
+function mapToSchoolExternalTool(externalTool, ltiToolTemplate, context) {
+	return {
+		tool: externalTool._id,
+		school: context.schoolId,
+		schoolParameters: mapToCustomParameterEntry(externalTool.parameters, ltiToolTemplate.customes),
```

Comment:
when migrating from lti to ctl the school parameters are always empty, since the school config layer does not exist in lti